### PR TITLE
#90 OCC via XML

### PR DESCRIPTION
**Pull request by llorllale**
As per #90 

> Optimistic Class Cohesion (OCC) quantifies the maximum extent
          of shared attributes between any two methods among all methods of
          a class. That is, for all pairs of methods of a class, it calculates
          a score based on the number of attributes of the class they both use
          in common, and then the maximum score is taken as the value for OCC.
          OCC is given a value of 0 for the exceptional case when the class has
          only one method.

See the `aman04.pdf` paper for details.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#168](https://github.com/yegor256/jpeek/issues/168) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#168](https://github.com/yegor256/jpeek/issues/168) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/421743197). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by amihaiemil** [comment]
@llorllale build is failing, pls have a look :)

**Comment by llorllale** [comment]
@amihaiemil This one is currently blocked by #172 as well

**Comment by yegor256** [comment]
@llorllale should be fine now

**Comment by llorllale** [comment]
@amihaiemil please review

**Comment by amihaiemil** [comment]
@llorllale Yes, I'll have a look over your PRs soon (give me 1, 2 days). I don't have time right now, sorry.

File: src/main/java/org/jpeek/App.java [hunk 1]
```diff
@@ -230,6 +231,14 @@ public void analyze() throws IOException {
                 )
             );
         }
+        if (this.params.containsKey("OCC")) {
```
Comment (amihaiemil) [inline src/main/java/org/jpeek/App.java]: @llorllale same problem as with the other PR...

File: src/test/java/org/jpeek/MetricsTest.java [hunk 2]
```diff
@@ -70,6 +70,8 @@
  * @checkstyle VisibilityModifierCheck (500 lines)
  * @checkstyle JavadocVariableCheck (500 lines)
  * @checkstyle MagicNumberCheck (500 lines)
+ * @todo #90:30min OCC metric: need to implement the rest of the test cases.
```
Comment (amihaiemil) [inline src/test/java/org/jpeek/MetricsTest.java]: @llorllale same question, what does "no time" mean? Had trouble with the other test cases, did they fail?

**Comment by llorllale** [comment]
@amihaiemil I addressed both these concerns in the other PR. Let me know

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by llorllale** [comment]
ping @yegor256

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by amihaiemil** [comment]
@llorllale conflic with MetricsTest here as well :D

**Comment by llorllale** [comment]
@amihaiemil fixed conflicts, please review

**Comment by amihaiemil** [comment]
@yegor256 good to merge again

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>The code reviewer found at least three problems in the code.
>Comments were mostly about design problems, not cosmetic issues.

Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm :)

**Comment by ypshenychka** [comment]
@amihaiemil Thanks.

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+270](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+760](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10304](https://www.0crat.com/u/yegor256)

**Pull request merged**